File: src/main/java/com/example/webshop/model/CartItem.java
package com.example.webshop.model;

import jakarta.persistence.*;
import lombok.*;

@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@EqualsAndHashCode(onlyExplicitlyIncluded = true)
@Table(name = "cart_items")
public class CartItem{

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	@EqualsAndHashCode.Include
	private Integer cartItemId;

	@ManyToOne(fetch = FetchType.LAZY)
	@JoinColumn(name = "cart_id")
	private Cart cart;

	@ManyToOne(fetch = FetchType.LAZY)
	@JoinColumn(name = "product_id")
	private Product product;

	private Integer quantity;

}

File: web/src/model/CartItem.js
class CartItem{
	constructor(data){
		this.id = data.id
		this.cartId = data.cartId
		this.productId = data.productId
		this.productName = data.productName
		this.productPrice = data.productPrice
		this.quantity = data.quantity
	}

	get total(){
		return this.productPrice*this.quantity
	}
}

export default CartItem
